pkg/install/hypershift.go

```diff
@@ -521,3 +575,110 @@ func getHyperShiftOperatorImage(im []byte) string {
 }
+
+func (c *UpgradeController) operatorImagesUpdated(im []byte, operatorDeployment appsv1.Deployment) bool {
+	different := false
+
+	hsOperatorContainer := operatorDeployment.Spec.Template.Spec.Containers[0]
+
+	imObj := &imageapi.ImageStream{}
+	if err := yaml.Unmarshal(im, imObj); err != nil {
+		c.log.Error(err, "failed to get image stream content: ", err.Error())
+		return false
+	}
+
+	c.log.Info("comparing hypershift operator images to the existing hypershift operator")
+	for _, tag := range imObj.Spec.Tags {
+		switch tag.Name {
+		case util.ImageStreamAgentCapiProvider:
+			if tag.From.Name != getContainerEnvVar(hsOperatorContainer.Env, util.HypershiftEnvVarImageAgentCapiProvider) {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamAgentCapiProvider))
+				different = true
+			}
+		case util.ImageStreamAwsCapiProvider:
+			if tag.From.Name != getContainerEnvVar(hsOperatorContainer.Env, util.HypershiftEnvVarImageAwsCapiProvider) {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamAwsCapiProvider))
+				different = true
+			}
+		case util.ImageStreamAwsEncyptionProvider:
+			if tag.From.Name != getContainerEnvVar(hsOperatorContainer.Env, util.HypershiftEnvVarImageAwsEncyptionProvider) {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamAwsEncyptionProvider))
+				different = true
+			}
+		case util.ImageStreamAzureCapiProvider:
+			if tag.From.Name != getContainerEnvVar(hsOperatorContainer.Env, util.HypershiftEnvVarImageAzureCapiProvider) {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamAzureCapiProvider))
+				different = true
+			}
+		case util.ImageStreamClusterApi:
+			if tag.From.Name != getContainerEnvVar(hsOperatorContainer.Env, util.HypershiftEnvVarImageClusterApi) {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamClusterApi))
+				different = true
+			}
+		case util.ImageStreamKonnectivity:
+			if tag.From.Name != getContainerEnvVar(hsOperatorContainer.Env, util.HypershiftEnvVarImageKonnectivity) {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamKonnectivity))
+				different = true
+			}
+		case util.ImageStreamKubevertCapiProvider:
+			if tag.From.Name != getContainerEnvVar(hsOperatorContainer.Env, util.HypershiftEnvVarImageKubevertCapiProvider) {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamKubevertCapiProvider))
+				different = true
+			}
+		case util.ImageStreamHypershiftOperator:
+			if tag.From.Name != hsOperatorContainer.Image {
+				c.log.Info(fmt.Sprintf("image(%s) has changed", util.ImageStreamHypershiftOperator))
+				different = true
+			}
+		}
+	}
+
+	return different
+}
+
+func getContainerEnvVar(envVars []corev1.EnvVar, imageName string) string {
+	for _, ev := range envVars {
+		if ev.Name == imageName {
+			return ev.Value
+		}
+	}
+	return ""
+}
+
+func (c *UpgradeController) secretDataUpdated(bucketSecret, privateLinkSecret, extDnsSecret corev1.Secret) bool {
```

Comment:
This code is pretty repetitive, maybe one func for get and compare, that you call 3x times or loop and call for each secret `type.NamespacedName`